platform/source/minimal-printf/mbed_printf_armlink_overrides.c

```diff
@@ -0,0 +1,375 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Arm Compiler uses dedicated functions for each format specifier used by
+ * [sn/v/vsn]printf. When minimal-printf overwrites [sn/v/vsn]printf the
+ * linker is unable to remove unused functions related to printf.
+ *
+ * The following stubs replace the built-in functions and helps the linker
+ * to resolve dependencies and correctly remove unused functions.
+ */
+
+#ifdef MBED_MINIMAL_PRINTF
+
+#if defined(TOOLCHAIN_ARM)
+
+#include "mbed_printf_implementation.h"
+
+#include <limits.h>
+#include <stdio.h>
+#include <stdarg.h>
+
+/**
+ * Arm Compiler uses __2[s/sn/vsn]printf internally.
+ */
+int $Sub$$__2printf(const char *format, ...)
```

Comment:
Why can't the functions here call the functions in `mbed_printf_wrapper.c`?

Reply:
You can't call the functions defined in `mbed_printf_wrapper.c` directly.  You need to declare a va_list variable first and then call a function that takes a va_list, which is what the mbed_minimal_formatted_string function does.

Reply:
Are we going to overlook the fact that the same code inside the functions appear on both files?

Reply:
Discussed with @hugueskamba that a further optimisation might be possible if compiler options (-Ono_fp_formats, --library_interface) for ARMC5 and ARMC6 exist but futher investigation would be necessary. 
Agreed that current code is fine.